Gordon360/Services/LostAndFoundService.cs

```diff
@@ -202,71 +246,104 @@ public async Task UpdateMissingItemReportAsync(int id, MissingItemReportViewMode
             }
         }
 
         /// <summary>
         /// Get the list of missing item reports for given user.
         /// </summary>
-        /// <param name="username">The username to get reports for</param>
+        /// <param name="username">The username of the person making the request</param>
         /// <returns>an Enumerable of Missing Item Reports containing all missing item reports</returns>
         public IEnumerable<MissingItemReportViewModel> GetMissingItems(string username)
         {
             return context.MissingItemData
                 .Where(x => x.submitterUsername == username && !x.forGuest)
                 .GroupJoin(context.ActionsTakenData
                 .Where(x => x.isPublic),
-                    mi => mi.ID,
-                    at => at.missingID, 
-                    (mi, at) => MissingItemReportViewModel.From(mi, at));
+                    missingItem => missingItem.ID,
+                    action => action.missingID,
+                    (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
         }
 
         /// <summary>
         /// Get all missing item reports
+        /// Throw unauthorized access exception if the user doesn't have admin permissions
         /// </summary>
+        /// <param name="username">The username of the person making the request</param>
         /// <returns>An enumerable of Missing Item Reports, from the Missing Item Data view</returns>
-        public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll()
+        /// <exception cref="UnauthorizedAccessException">If a username without admin permissions attempts to use</exception>
+        public IEnumerable<MissingItemReportViewModel> GetMissingItemsAll(string username)
         {
-            // Perform a group join to create a MissingItemReportViewModel with actions taken data for each report
-            // Only performs a single SQL query to the db, so much more performant than alternative solutions
-            return context.MissingItemData
-                .GroupJoin(context.ActionsTakenData,
-                    missingItem => missingItem.ID, 
-                    action => action.missingID, 
-                    (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
+            if (hasFullPermissions(username))
+            {
+                // Perform a group join to create a MissingItemReportViewModel with actions taken data for each report
+                // Only performs a single SQL query to the db, so much more performant than alternative solutions
+                return context.MissingItemData
+                    .GroupJoin(context.ActionsTakenData,
+                        missingItem => missingItem.ID,
+                        action => action.missingID,
+                        (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
+            }
+            throw new UnauthorizedAccessException();
```

Comment:
This is just a stylistic choice, but I think code like is easier to read when you handle the exceptional cases early and leave the main logic "unindented":
```suggestion
            if (!hasFullPermissions(username))
            {
                throw new UnauthorizedAccessException();
            }
            
            // Perform a group join to create a MissingItemReportViewModel with actions taken data for each report
            // Only performs a single SQL query to the db, so much more performant than alternative solutions
            return context.MissingItemData
                .GroupJoin(context.ActionsTakenData,
                    missingItem => missingItem.ID,
                    action => action.missingID,
                    (missingItem, action) => MissingItemReportViewModel.From(missingItem, action));
```

It doesn't matter as much in a method this small, but it means you have less cognitive load trying to remember all the cases. I.e. in the current version, you have to remember what the condition was after you finished looking at the main groupjoin to remember whether and when the exception gets thrown. In the version I'm proposing, you can verify that the condition makes sense looking at just the condition and it's exception, and then forget about that case and focus on the main idea of the code, which is the SQL query and data formatting. You can imagine how that would add up if you had 5 or 6 exceptional cases to handle, where one version of code lets you verify and then forget the exceptions, while the other requires you to keep all the conditions and edge cases in mind throughout the whole method to understand what will execute when.